scripts/stablex/place_margin_orders.js

```diff
@@ -0,0 +1,91 @@
+const BatchExchange = artifacts.require("BatchExchange")
+const { sendTxAndGetReturnValue } = require("../../test/utilities.js")
+const BN = require("bn.js")
+
+const argv = require("yargs")
+  .option("tokens", {
+    describe: "Collection of trusted tokens",
+  })
+  .option("accountId", {
+    describe: "Account index of the order placer",
+  })
+  .option("margin", {
+    type: "float",
+    describe: "Percentage increase required for trade (fees not accounted)",
+    default: 0.2,
+  })
+  .option("sellAmount", {
+    type: "float",
+    describe: "Maximum sell amount (considered infinite if empty)",
+    default: 100,
+  })
+  .option("expiry", {
+    type: "int",
+    describe: "Maximum auction batch for which these orders are valid",
+    default: 2**32 - 1
+  })
+  .demand(["tokens", "accountId"])
+  .help(false)
+  .version(false).argv
+
+
+// TODO - automate this with tokenIdToAddress dot decimals
+const TOKEN_DECMIALS = {
+  2: 6,
+  3: 18,
+  4: 6,
+  5: 18,
+  6: 2,
+  7: 18
+}
+
+module.exports = async callback => {
+  try {
+    const instance = await BatchExchange.deployed()
+    const accounts = await web3.eth.getAccounts()
+    const account = accounts[argv.accountId]
+    console.log(account)
+    const batch_index = (await instance.getCurrentBatchId.call()).toNumber()
+
+    const trusted_tokens = argv.tokens.split(",").map(t => parseInt(t))
+    const expectedReturnFactor = new BN(1 + argv.margin / 100)
+    const sellAmount = new BN(argv.sellAmount)
+    const buyAmount = expectedReturnFactor.mul(sellAmount)
+
+    let buyTokens = []
+    let sellTokens = []
+    let buyAmounts = []
+    let sellAmounts = []
+    for (let i = 0; i < trusted_tokens.length - 1; i++) {
+      const tokenA = trusted_tokens[i]
+      const tokenScaleA = new BN(10).pow(new BN(TOKEN_DECMIALS[tokenA]))
```

Comment:
there's web3 utils for this: fromWei, toWei

if you use that, you just have to declare `tokenAInWei` instead of creating the scales

Reply:
Yes cool! Thanks I’ll look into that! Does it take the number of decimals as a parameter as well?

Reply:
Unfortunately @anxolin - web3's `toWei` only accepts these stupid word parameters like "finney" and I can't simply supply the number of decimals like a regular person would desire.

https://web3js.readthedocs.io/en/v1.2.0/web3-utils.html#towei

Reply:
Yep, now that you say it, I knew this. I had to do a util for formatting weis in the frontend.

Anyways, I recommend you to create a function yourself, so it's easier to read and you can use it for any other script.